site-manage/site-group-manager/src/main/java/org/sakaiproject/groupmanager/controller/MainController.java

```diff
@@ -0,0 +1,145 @@
+import java.util.stream.Collectors;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.sakaiproject.authz.api.AuthzGroup.RealmLockMode;
+import org.sakaiproject.authz.api.AuthzRealmLockException;
+import org.sakaiproject.groupmanager.constants.GroupManagerConstants;
+import org.sakaiproject.groupmanager.form.MainForm;
+import org.sakaiproject.groupmanager.service.SakaiService;
+import org.sakaiproject.site.api.Group;
+import org.sakaiproject.site.api.Site;
+import org.sakaiproject.site.util.SiteComparator;
+import org.sakaiproject.site.util.SiteConstants;
+import org.sakaiproject.user.api.User;
+
+@Slf4j
+@Controller
+public class MainController {
+    
+    @Autowired
+    private SakaiService sakaiService;
+
+    @RequestMapping(value = {"/", "/index"})
+    public String showIndex(Model model) {
+        log.debug("showIndex()");
+
+        Site site = sakaiService.getCurrentSite();
+        if(site == null) {
+        	return GroupManagerConstants.REDIRECT_MAIN_TEMPLATE;
+        }
+
+        // Group members for each group, separated by comma
+        Map<String, String> groupMemberMap = new HashMap<String, String>();
+        // Joinable sets for each group
+        Map<String, String> groupJoinableSetMap = new HashMap<String, String>();
+        // Joinable sets size for each group
+        Map<String, String> groupJoinableSetSizeMap = new HashMap<String, String>();
+        // List of groups of the site, excluding the ones which GROUP_PROP_WSETUP_CREATED property is false.
+        List<Group> groupList = site.getGroups().stream().filter(group -> group.getProperties().getProperty(Group.GROUP_PROP_WSETUP_CREATED) != null && Boolean.valueOf(group.getProperties().getProperty(Group.GROUP_PROP_WSETUP_CREATED)).booleanValue()).collect(Collectors.toList());
+        // Sort the group list by title.
+        Collections.sort(groupList, new Comparator<Group>(){
+            public int compare(Group g1, Group g2){
+                return g1.getTitle().compareToIgnoreCase(g2.getTitle());
+        }});
+
+        List<Group> lockedGroupList = site.getGroups().stream().filter(group -> RealmLockMode.ALL.equals(group.getRealmLock()) || RealmLockMode.MODIFY.equals(group.getRealmLock())).collect(Collectors.toList());
+        List<Group> lockedForDeletionGroupList = site.getGroups().stream().filter(group -> RealmLockMode.ALL.equals(group.getRealmLock()) || RealmLockMode.DELETE.equals(group.getRealmLock())).collect(Collectors.toList());
+
+        // For each group of the site, get the members separated by comma, the joinable sets and the size of the joinable sets.
+        for(Group group: groupList) {
```

Comment:
tut tut

Reply:
Eclipse complains about the types, I'll leave them.